src/TriggerService/TriggerHostedService.cs

```diff
@@ -327,11 +331,27 @@ public async Task<ProcessedWorkflowItem> GetBlobFileNameAndData(string url)
 
             if (GetBlockBlobStorage(url, storageAccounts) is IAzureStorage aStorage)
             {
-                data = await aStorage.DownloadBlockBlobAsync(url);
+                try
+                {
+                    data = await aStorage.DownloadBlockBlobAsync(url);
+                }
+                catch (Exception e)
+                {
+                    throw new Exception($"{e.InnerException} while processing blob, cannot find {url} in {aStorage.AccountName}. " +
+                        $"Please confirm blob is accessible and named correctly before resubmitting your trigger file.");
+                }
             }
             else
             {
-                data = await storage.DownloadFileUsingHttpClientAsync(url);
+                try
+                {
+                    data = await storage.DownloadFileUsingHttpClientAsync(url);
+                }
+                catch (Exception e)
+                {
+                    throw new Exception($"{e.InnerException} while downloading file from Http url {url}. " +
```

Comment:
e?.InnerException in case it's null.  "Http url" should be HTTP URL for proper capitalization